jailer.go

```diff
@@ -288,7 +290,18 @@ func jail(ctx context.Context, m *Machine, cfg *Config) error {
 		jailerWorkspaceDir = filepath.Join(defaultJailerPath, filepath.Base(cfg.JailerCfg.ExecFile), cfg.JailerCfg.ID, rootfsFolderName)
 	}
 
-	cfg.SocketPath = filepath.Join(jailerWorkspaceDir, "run", "firecracker.socket")
+	var machineSocketPath string
+	if cfg.SocketPath != "" {
+		machineSocketPath = cfg.SocketPath
+	} else {
+		machineSocketPath = defaultSocketPath
+	}
+
+	cfg.SocketPath = filepath.Join(jailerWorkspaceDir, machineSocketPath)
+
+	if err := os.MkdirAll(filepath.Dir(cfg.SocketPath), 0600); err != nil {
```

Comment:
This needs to be done in a separate handler due to Firecracker creating the initial jailer workspace. You could probably just do it in the `LinkFilesHandler`, for now.

Reply:
@xibz I thought about it, but the LinkFilesHandler does not have access to the config. It only knows about what was passed to the NaiveChrootStrategy. I would rather not have to add another argument to that constructor, especially since it's repeating the SocketPath.

Open to suggestions!

Reply:
It does have access to the config, luckily. I actually prefer it in the handler, so if we could move it there, that'd be awesome.

The [handler](https://github.com/firecracker-microvm/firecracker-go-sdk/blob/master/handlers.go#L280) is a closure that passes the `Machine` into it. From there you should be able to access the config.

Reply:
Actually, after some thought, I prefer this in the jail function. So this is fine as is. Ill take a look at why this is failing in our CI